# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/sparc64-sigaction.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Sigaction on sparc needs a return stub, compare with glibc and latest klibc comments
from David S. Miller.

  - Rene Rebe <rene@exactcode.de>

--- dietlibc-0.29-t2/sparc64/sigaction.c	1970-01-01 00:00:00.000000000 +0000
+++ dietlibc-0.29/sparc64/sigaction.c	2006-06-18 10:42:49.000000000 +0000
@@ -0,0 +1,21 @@
+#include <signal.h>
+#include "syscalls.h"
+
+static void
+__rt_sigreturn_stub (void)
+{
+  __asm__ ("mov %0, %%g1\n\t"
+           "ta  0x6d\n\t"
+           : /* no outputs */
+           : "i" (__NR_rt_sigreturn));
+}
+
+int __rt_sigaction(int signum, const struct sigaction *act, struct sigaction *oldact, void* restorer, long nr);
+
+int __libc_sigaction(int signum, const struct sigaction *act, struct sigaction *oldact);
+int __libc_sigaction(int signum, const struct sigaction *act, struct sigaction *oldact) {
+  return __rt_sigaction(signum, act, oldact, (void*)(&__rt_sigreturn_stub) - 8, _NSIG/8);
+}
+int sigaction(int signum, const struct sigaction *act, struct sigaction *oldact)
+__attribute__((weak,alias("__libc_sigaction")));
+
